### PR TITLE
Pass the host route IP family to the guest 

### PR DESCRIPTION
**Pull request by sameo**
So that we can properly select the same IP family when setting and updating routes in the guest.

Fixes: #2768

### CODE REVIEW

**Comment by fidencio** [comment]
Although this sounds intrusive for a backport, I think it's needed & worth it.  Could someone from @kata-containers/redhat review this one?

**Comment by sameo** [comment]
> Although this sounds intrusive for a backport, I think it's needed & worth it. Could someone from @kata-containers/redhat review this one?

Right, I will backport it once it's approved and CI green.

File: src/agent/src/netlink.rs
Comment (egernst) [inline src/agent/src/netlink.rs:316]: Is there any reason we wouldn't want to use get_family directly, regardless if the gateway and destination are not provided?  Are we populating the family as part of the gRPC regardless?

Comment (sameo) [reply src/agent/src/netlink.rs:316]: This is for backward compatibility purposes, unless we're ok with saying the new agent will only work with the corresponding  runtime changes.

File: src/runtime/cmd/netmon/netmon.go
Comment (c3d) [inline src/runtime/cmd/netmon/netmon.go:173]: Is this a change in the interface? If so, could you announce it in the commit message to explain the change?

**Review by c3d — APPROVED** [review]
Looks good to me.

Do you have a test case? (I did not see it)

File: src/runtime/cmd/netmon/netmon.go
Comment (sameo) [reply src/runtime/cmd/netmon/netmon.go:173]: Good point. I added a note on the commit message now.

File: src/agent/src/netlink.rs
Comment (sameo) [reply src/agent/src/netlink.rs]: @egernst I changed my mind and went the cleaner way: We only rely on the Route family field now for switching between v4 and v6. Running this new agent means you'll have to run the fixed runtime as well, but I believe this is acceptable.

**Review by Jakob-Naucke — APPROVED** [review]

**Review by jodh-intel — APPROVED** [review]
Thanks @sameo.

lgtm. Since @egernst has commented, maybe we should hold off merging until he's acked this too.

File: src/runtime/go.sum
Comment (gkurz) [inline src/runtime/go.sum:535]: Shouldn't the go.mod line be removed all the same ?

Comment (sameo) [reply src/runtime/go.sum:535]: FWIW, this was autogenerated for me by `go mod`. I'm not sure I want to manually modify that file...

Comment (gkurz) [reply src/runtime/go.sum:535]: Indeed ! And if you remove it, `go mod tidy` consistently puts it back. Thanks, that answers my question.

**Pull request merged**
